### PR TITLE
Added client method to get tracks.  Added "--track_promote_to" option…

### PR DESCRIPTION
**Pull request by sschendel**
… to add capability to promote apk version codes between tracks.  Proposed implementation of feature in issue #48.

### CODE REVIEW

**Comment by asfalcone** [comment]
@sschendel Thanks for the PR. This is a great enhancement. One thing to note is that there is an outstanding PR #46 to update to the newer google client API, which will be merged first. I believe this PR is going to need to be rebased and fixed up after that PR gets merged. We've got a bit of a backlog, but we'll take a look at this enhancement we get a chance. Thanks!

**Comment by asfalcone** [comment]
@sschendel #46 has been merged. Could you please rebase and re-add the `track_promote_to` action? Thanks!

**Comment by sschendel** [comment]
@asfalcone Will do, but it may be a few days.

File: lib/supply/uploader.rb [hunk 1]
```diff
@@ -24,11 +24,23 @@ def perform_upload
...
+
       Helper.log.info "Uploading all changes to Google Play..."
       client.commit_current_edit!
       Helper.log.info "Successfully finished the upload to Google Play".green
     end
 
+    def promote_track
+      if Supply.config[:track_promote_to]
+        version_codes = client.track_version_codes(Supply.config[:track])
+        client.update_track(Supply.config[:track], 1.0, nil)
```
Comment (lacostej) [inline lib/supply/uploader.rb:37]: maybe we want to extract the following code
https://github.com/fastlane/supply/blob/0.4.0/lib/supply/uploader.rb#L93-L96
into one method and reuse it here (to get logging and rollout parameter support)

File: lib/supply/uploader.rb [hunk 2]
```diff
@@ -24,11 +24,23 @@ def perform_upload
...
+        version_codes.each do |apk_version_code|
+          client.update_track(Supply.config[:track_promote_to], 1.0, apk_version_code)
```
Comment (lacostej) [inline lib/supply/uploader.rb:39]: ditto

File: lib/supply/uploader.rb [hunk 3]
```diff
@@ -24,11 +24,23 @@ def perform_upload
...
+      promote_track
+
       Helper.log.info "Uploading all changes to Google Play..."
       client.commit_current_edit!
       Helper.log.info "Successfully finished the upload to Google Play".green
     end
 
+    def promote_track
+      if Supply.config[:track_promote_to]
+        version_codes = client.track_version_codes(Supply.config[:track])
```
Comment (lacostej) [inline lib/supply/uploader.rb:36]: Do we really want to promote all versions here ?

**Comment by sschendel** [comment]
@lacostej Appreciate all the feedback.  I'm refactoring my updates following rebase post PR #46 merge.

**Comment by lacostej** [comment]
@sschendel  cool ! Thanks :)

**Comment by asfalcone** [comment]
@sschendel Just wanted to ping on this PR again. We're moving to the monorepo in the coming weeks and would like to get this PR in. It would be great if you could address the feedback and rebase on master so we can merge this PR soon! 🔔

**Comment by sschendel** [comment]
@asfalcone Sorry, I'm not sure I'll be able to get to it in a timely manner.  My excuse: 5 month old infant twins.

**Comment by asfalcone** [comment]
Thanks @sschendel! After we move over to monorepo I'll try to reimplement this myself (or will happily accept another PR with these changes reworked against the updated google-api-client)!

**Comment by asfalcone** [comment]
I'm going to close this PR for now since it cannot be merged as is.

**Pull request closed without merging**